warp10/src/main/java/io/warp10/continuum/gts/MetadataSelectorMatcher.java

```diff
@@ -278,21 +278,23 @@ public static MetadataIterator getIterator(final Iterator<Metadata> iter, Direct
         Map<String,String> labels = dr.getLabelsSelectors().get(i);
         boolean first = true;
         for (Entry<String,String> entry: labels.entrySet()) {
-          if (!first) {
-            sb.append(",");
-          }
-          sb.append(WarpURLEncoder.encode(entry.getKey(), StandardCharsets.UTF_8));
-          if (entry.getValue().startsWith("=")) {
-            sb.append("=");
-            sb.append(WarpURLEncoder.encode(entry.getValue().substring(1), StandardCharsets.UTF_8));
-          } else if (entry.getValue().startsWith("~")) {
-            sb.append("~");
-            sb.append(WarpURLEncoder.encode(entry.getValue().substring(1), StandardCharsets.UTF_8));
-          } else {
-            sb.append("=");
-            sb.append(WarpURLEncoder.encode(entry.getValue(), StandardCharsets.UTF_8));
+          if (entry.getValue() != null) { // it could be null (.owner, for example)
```

Comment:
null != entry.getValue()

Could you explain in more details when getValue could return null, it is not clear from the current comment

Reply:
I tried to understand the origin of .owner=null , but I cannot figure it out. 
- It is not in FIND.java line 742.
- There is a owner key in the read token I used to test.